iml-manager-cli/src/main.rs

```diff
@@ -172,20 +191,60 @@ fn main() {
                             purge_duration: pd
                         }),
                     )
+                    .and_then(|resp| iml_manager_client::concat_body(resp))
+                    .map(|(resp, chunk)| {
```

Comment:
Instead of using `map` here, using `and_then` would allow the Error to propagate to the error side of the channel.

Then the Result is lifted up to the future.